### PR TITLE
Password Policy: add documentation

### PR DESCRIPTION
**Pull request by linoman**
**What is this feature?**

Add documentation for the new password policy feature.

**Why do we need this feature?**

To keep documentation up to date.

**Who is this feature for?**

Everyone.

**Which issue(s) does this PR fix?**:

Fixes #https://github.com/grafana/identity-access-team/issues/573

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: docs/sources/setup-grafana/configure-security/configure-authentication/grafana/index.md [hunk 1]
```diff
@@ -104,6 +104,23 @@ To disable basic auth:
 enabled = false
 ```
 
+### Strong password policy
+
+By default, the password policy for all basic auth users is set to a minimum of 4 characters. You can enable a stronger password policy with the `password_policy` configuration option.
+
+With the `password_policy` option enabled, new and updated passwords must meet the following criteria:
```
Comment (IevaVasiljeva) [inline docs/sources/setup-grafana/configure-security/configure-authentication/grafana/index.md:111]: Is it worth calling out that the existing passwords will not be impacted?

**Review by IevaVasiljeva — APPROVED** [review]
LGTM content wise! Could you also add the new setting to this doc page: https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#auth 🙏

**Comment by linoman** [comment]
> Could you also add the new setting to this doc page: https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#auth 🙏

@IevaVasiljeva I'm not sure what you meant. The sections for each of the different auth mechanisms are only redirects to the pages of the specific mechanisms themselves.

File: docs/sources/setup-grafana/configure-security/configure-authentication/grafana/index.md [hunk 2]
```diff
@@ -104,6 +104,27 @@ To disable basic auth:
 enabled = false
 ```
 
+### Strong password policy
+
+By default, the password policy for all basic auth users is set to a minimum of 4 characters. You can enable a stronger password policy with the `password_policy` configuration option.
```
Comment (chri2547) [inline docs/sources/setup-grafana/configure-security/configure-authentication/grafana/index.md:109]:
```suggestion
By default, the password policy for all basic auth users is set to a minimum of four characters. You can enable a stronger password policy with the `password_policy` configuration option.
```

**Review by chri2547 — APPROVED** [review]
One very small nit, looks great! Thank you for the contribution!

**Pull request merged**